test/defaultSnippets.test.ts

```diff
@@ -28,7 +28,23 @@ describe('Default Snippet Tests', () => {
   });
 
   describe('Snippet Tests', function () {
-    function parseSetup(content: string, position: number): Promise<CompletionList> {
+    /**
+     * Generates a completion list for the given document and caret (cursor) position.
+     * @param content The content of the document.
+     * @param position The position of the caret in the document.
+     * Alternatively, `position` can be omitted if the caret is located in the content using `|` bookends.
+     * For example, `content = 'ab|c|d'` places the caret over the `'c'`, at `position = 2`
+     * @returns A list of valid completions.
+     */
+    function parseSetup(content: string, position?: number): Promise<CompletionList> {
+      // console.log(`was: len: ${content.length}, content: "${content}", str: "${content.substring(position)}"`);
+      if (typeof position === 'undefined') {
+        position = content.search(/\|[^]\|/); // | -> any char including newline -> |
```

Comment:
How about throwing an error of caret is not found. Perhaps move this 2 lines to its own function on test\utils\testHelper.ts?

Reply:
Done